router/health_check.go

```diff
@@ -0,0 +1,123 @@
+package router
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/buildwithgrove/path/relayer/morse"
+	"github.com/buildwithgrove/path/relayer/shannon"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+// HealthCheck is an interface that must be implemented
+// by components that need to report their health status
+type HealthCheck interface {
+	// Name returns the name of the component being checked.
+	Name() string
+	// IsAlive returns true if the component is healthy, otherwise false.
+	IsAlive() bool
+}
+
+// All components that implement the HealthCheck interface must be registered here.
```

Comment:
Please move this into the corresponding files where the interface is being implemented.

It's a way to tell the developer: "You didn't implement the interface correctly".

The library (i.e. where the interface is defined) shouldn't have knowledge of who is implementing git.

Reply:
Not possible. It creates an import cycle between `router` and `shannon/morse`; that's why I put it where I did.

Reply:
[Solved this by moving the health check to its own package.](https://github.com/buildwithgrove/path/pull/20#discussion_r1764602485)